### PR TITLE
[Feature] Add filters to screening and assessment page

### LINKED ISSUE #9948: ✨ Add filters to 'Screening and assessment' page
## User story
As a pool operator, I want to filter the screening and assessment table, so I can focus on (for example) just EE candidates first.

As a pool operator, I want to sort candidates by applications date, so I can assess them in the order that they applied.

## Details
Add filter button to the screening and assessment page.

## Design
@Jerryescandon 
https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Admin---Processes%2FPools%2FRecruitments---Record-of-Decision?type=design&node-id=8738%3A33892&mode=design&t=55lrreznUEbI4ioU-1

##  Screenshots
image
image

## Acceptance criteria

> [!IMPORTANT]
> Sorting is not part of the scope for this ticket. It may be  introduced in the future but for now, do not include it.

- [ ] Filter ~~and sort~~ button added to Screening and assessment page
- [ ] Filter ~~and sort~~ dialogue has all the same options as the designs (ok if they are implemented a little different though)
- [ ] Filtering ~~and sort~~ actually work on the Screening and assessment kanban thingy

🛑 Blockers
```[tasklist]
### Blocked by
- [ ] #8966
```

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #9948 

## 👋 Introduction

- Adds filter dialog to Screening and assessment page
- Adds the filters specified in design
- Filters out candidates that are expired or suspended by default. 

## 🧪 Testing

1. Build app `pnpm run dev`
2. Login as `admin@test.com`
3. Go to process and view screening and assessment page
4. Ensure the dialog filters out the correct candidates. Ensure the other filters still work correctly.

## 📸 Screenshot

image

### CODE REVIEW

**Comment by Jerryescandon** [comment]
I think it should be fine to have them be filtered out by default as long as we can go into the filters to show them again.

**Review by esizer — CHANGES_REQUESTED** [review]
Works well! :+1:  I just have a few suggestions.

File: apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    }
+    suspendedFilters: localizedEnumStrings(
+      enumName: "CandidateSuspendedFilter"
+    ) {
+      value
+      label {
+        en
+        fr
+      }
+    }
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx]: How did I miss this :laughing: Fixed by [8c68e43](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11018/commits/8c68e439a9ddcb7cdac3e4a46fcf2a792d48c112)

File: apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+            idPrefix="suspendedStatus"
+            name="suspendedStatus"
+            legend={intl.formatMessage({
+              defaultMessage: "Candidacy status",
+              description: "Label for the candidacy status field",
+              id: "NxrKpM",
+            })}
+            items={localizedEnumToOptions(data?.suspendedFilters, intl)}
+          />
+        </div> */}
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx]: This is not getting used I added it while it was undetermined whether these fields would exist on dialog  :laughing: Fixed by [8c68e43](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11018/commits/8c68e439a9ddcb7cdac3e4a46fcf2a792d48c112)

File: apps/web/src/components/AssessmentStepTracker/AssessmentStepTracker.tsx [hunk 3]
```diff
@@ -127,7 +133,26 @@ const AssessmentStepTracker = ({
...
+        data-h2-display="base(flex)"
+        data-h2-align-items="base(flex-end)"
+        data-h2-gap="base(x.5)"
+        data-h2-margin-bottom="base(x1) l-tablet(x.25)"
+        data-h2-justify-content="base(flex-start)"
+        data-h2-font-size="base(caption)"
+        data-h2-flex-wrap="base(wrap)"
+      >
+        <Filters onFiltersChange={setFilters} />
+        <div data-h2-margin-bottom="l-tablet(x.5)">
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/components/AssessmentStepTracker/AssessmentStepTracker.tsx]: I tried to copy the styling for the other tables. If this doesn't suffice then I can try something else.
[c4309c2](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11018/commits/c4309c2c8d8515153f577bd8e8f87f3b8b2fde90)

File: apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+            id: "iUAe/2",
+            description: "Label for language ability field",
+          })}
+          options={localizedEnumToOptions(data?.languageAbilities, intl)}
+        />
+        <Select
+          id="equity"
+          name="equity"
+          enableNull
+          nullSelection=""
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx]: Fixed by [890dec2](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11018/commits/890dec2bdc74d908bc829b0dfec81b729fec9756)

File: apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+            value: id,
+            label: getLocalizedName(name, intl),
+          }))}
+        />
+        <div data-h2-grid-column="l-tablet(span 2)">
+          <Select
+            id="operationalRequirement"
+            name="operationalRequirement"
+            enableNull
+            nullSelection=""
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/components/AssessmentStepTracker/AssessmentResultsFilterDialog.tsx]: Fixed by [e03e60e](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11018/commits/e03e60e84bec96b7fbcb84d6867bf201d2c2571d)

**Review by esizer — APPROVED** [review]

**Pull request merged**